ui/app/src/components/PreviewSearchPanel/PreviewSearchPanel.tsx

```diff
@@ -139,52 +147,61 @@ export function PreviewSearchPanel() {
     (keywords: string = '', options?: ComponentsContentTypeParams) => {
       setState({ isFetching: true });
       setError(null);
-      search(site, {
-        ...initialSearchParameters,
-        keywords,
-        ...options,
-        filters: { 'content-type': contentTypes?.map((item) => item.id), 'mime-type': mimeTypes }
-      })
-        .pipe(
-          takeUntil(unMount$),
-          switchMap((result) => {
-            const requests: Array<Observable<ContentInstance>> = [];
-            result.items.forEach((item) => {
-              if (item.type === 'Component') {
-                requests.push(fetchContentInstance(site, item.path, contentTypesLookup));
-              }
-            });
-            return requests.length
-              ? forkJoin(requests).pipe(map((contentInstances) => ({ contentInstances, result })))
-              : of({ result, contentInstances: null });
-          })
-        )
-        .subscribe({
-          next: (response) => {
-            setPageNumber(options ? options.offset / options.limit : 0);
-            if (response.contentInstances) {
-              setState({
-                isFetching: false,
-                items: response.result.items,
-                contentInstanceLookup: createLookupTable(response.contentInstances, 'craftercms.path'),
-                count: response.result.total,
-                limit: options?.limit ?? initialSearchParameters.limit
-              });
-            } else {
-              setState({
-                isFetching: false,
-                items: response.result.items,
-                count: response.result.total,
-                limit: options?.limit ?? initialSearchParameters.limit
+      if (allowedTypesData) {
+        // If there are no allowed types, an empty array causes the request to return everything.
+        const allowedTypes = Object.keys(allowedTypesData).length
+          ? Object.entries(allowedTypesData)
+              .filter(([, type]) => type.shared)
+              .map(([key]) => key)
+          : [''];
```

Comment:
Not very efficient or performant to make the engine parse the keys and also parse the entries
```suggestion
        // If there are no allowed types, an empty array causes the request to return everything.
        let allowedTypes = Object.entries(allowedTypesData).filter(([, type]) => type.shared)
              .map(([key]) => key);
       if (allowedTypes.length === 0) allowedTypes = ['']
```

Reply:
If there are no allowed types, don't we want to get assets only, instead of getting everything? Perhaps the comment needs fixing?

Reply:
Updated the comment to be more detailed. Also, allowed types is only for the `content-type` filter, `mime-type` stays the same so assets will be retrieved.